### ISSUE TITLE
slab Hamiltonian for top and bottom surfaces

### ISSUE THREAD

**Issue by elementh11** [quanshengwu/wannier_tools#45]
Is it possible to build a slab Hamiltonian (with a construction like in Eq. 10 in Comp. Phys. Comm. 242, 405) that will separately describe the top and bottom surfaces?
I would like to diagonalize H_{slab,top} and H_{slab, bottom} to get E vs k for each surface...

**Comment by quanshengwu** [comment]
Yes,  It is possible. However, you have to diagonalize H_slab{both surface} first. Then select the surface states for top or bottom according to the surface projections. Last, you need to project the H_slabb{both surface} onto the surface state eigenvectors to get an effective model.  In principle, this is the same as the procedure to deduce surface Hamiltonian from the bulk k.p model.

**Comment by elementh11** [comment]
Thank you for your reply. I'm not sure I understand how to separate the top and bottom surface states.
After I diagonalize H_slab{both surface}, how do I know which eigenvalues correspond to the top surface and which ones to the bottom surface?
